node/src/actors/json_rpc/json_rpc_methods.rs

```diff
@@ -1098,17 +1098,30 @@ pub async fn data_request_report(params: Result<(Hash,), jsonrpc_core::Error>) -
         .await
 }
 
+/// Params of getBlockChain method
+#[derive(Debug, Default, Deserialize, Serialize)]
+pub struct GetBalanceParams {
+    /// Public key hash
+    #[serde(default)] // default to default pkh
+    pub pkh: PublicKeyHash,
+    /// Distinguish between fetching a simple balance or fetching confirmed and unconfirmed balance
+    #[serde(default)] // default to false
+    pub simple: bool,
+}
+
 /// Get balance
-pub async fn get_balance(params: Result<(PublicKeyHash,), jsonrpc_core::Error>) -> JsonRpcResult {
-    let pkh = match params {
-        Ok(x) => x.0,
+pub async fn get_balance(
+    params: Result<Option<GetBalanceParams>, jsonrpc_core::Error>,
```

Comment:
Changing the input type is technically a breaking change, but it may not be a problem because getBalance is only used in the CLI, and we may break the JSON-RPC API in PR #1939 anyway.

But if we want to make this backwards-compatible, this method should take an unparsed `Params`, similar to the `get_block` method, and it should parse the input manually: if params is an array, parse using old logic and set simple to false; if params is an object parse using new logic.

```suggestion
    params: Params,
```